text/0038-scrubbing-sensitive-data.md

```diff
@@ -0,0 +1,136 @@
+- Start Date: 2022-11-21
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/38
+
+# Summary
+
+Make scrubbing of sensitive data (security related data (passwords, keys, et al) and personal identifiable information (PII)) in Sentry smarter.
+This includes the SDK side (before we send data) and the Relay side (before we ingest data)
+
+# Motivation
+
+Currently the scrubbing of sensitive data works, but is not very smart. It can happen that sensitive data is leaked and ends up in our data storage.
+It also can happen, that scrubbing is too broadly defined and too much data is removed so the feature distroys value for the customer.
+
+We want to have a smarter way of scrubbing sensitive data with a more fine grained control of what data is preserved and what data is scrubbed.
+
+We do this to make sure we handle our users sensitive data with the respect it deserves.
+
+# Background
+
+A user complained that our data scrubbing did not remove sensitive data from an URL the users project made a HTTP request to. The sensitive data showed up in the span description in the Sentry UI.
+
+# Supporting Data
+
+(tbd)
+
+# Conclusion
+
+Not one option will be implemented but a combination of options:
+
+1. The SDKs will try to save structured data where possible (Option B)
+2. The SDKs will try to specify what kind of data is in `span.description` (and other places) as good as possible. (Can be done by more fine grained `span.op`s or applying OTel trace semantic convention to `span.data`. Needs to be speced out)
```

Comment:
`h`: If the SDK already uses structured data, should it also specify what kind of data is in `span.description`? If yes, how do we prevent Relay from changing the already structured data?